sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_floating_point.asciidoc

```diff
@@ -0,0 +1,1280 @@
+|`OpArbitraryFloatCastToIntINTEL`         | 5843
+|====
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16a Subnormal Support" as follows:
+
+=== Subnormal Support
+
+Control whether subnormal support is enabled or not.
+
+[options="header"]
+[width="60%"]
+[cols="15%,85%"]
+|====
+| Value ^| Subnormal Support (`OpTypeInt`)
+| 0 | Flush subnormals numbers to zero on inputs and outputs
+| 1 | Enable support for operating on subnormal numbers
+|====
+
+After Section 3.16, add a new section "3.16d Rounding Accuracy" as follows:
+
+=== Rounding Accuracy
+
+Controls whether rounding operations can be relaxed to trade correctness for improved resource utilization.
+
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | CORRECT   | Conform to the IEEE-754 standard for rounding accuracy
+| 1      | FAITHFUL  | Allow some tolerance for error (within 1ULP of the IEEE-754 standard) for rounding. +
+The returned result is one of the two floating point values closest to the mathematical result.
+
+This mode is useful for devices that can trade CORRECT rounding for improved resource utilization.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5845 | *ArbitraryPrecisionFloatingPointINTEL* +
+
+Allows the use of various operations for arbitrary precision floating-point math
+
+| *ArbitraryPrecisionIntegersINTEL* |
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="14", width="100%"]
+|=====
+13+<|*OpArbitraryFloatAddINTEL* +
+
+Two `OpTypeInts` representing two arbitrary precision floating point numbers (ihc_float) are passed in as _A_ and _B_ and the result of `A+B` is returned in _Result_.
+
+_Result Type_ is an `OpTypeInt` of width `1+Eout+Mout` and is the type of _Result_.
+
+_A Type_ is an `OpTypeInt` of width `1+E1+M1` and is the type of _A_.
+
+_B Type_ is an `OpTypeInt` of width `1+E2+M2` and is the type of _B_.
+
+_E*_ and _M*_ contain the width of the exponent and the mantissa of the floating point types within _Result_, _A_, and _B_.
+Note that the exponent values (E1, E2, Eout) are inferred from the width of the `OpTypeInt`.
+
+_EnableSubnormals_ specifies whether subnormal numbers should be supported or flushed to zero before and after the operation and is chosen from _Table 3.16a_.
+
+_RoundingMode_ controls the rounding mode for the result and is chosen from _Table 3.16_.
+
+_RoundingAccuracy_ controls the accuracy of the rounding operation and is chosen from _Table 3.16d_.
+
+| Capability:
+*ArbitraryPrecisionFloatingPointINTEL*
+| 13 | 5846 | <id> Result Type | Result <id> | <id> A Type | A <id> | _Literal_ M1 | <id> B Type | B <id> | _Literal_ M2 | _Literal_ Mout | _Literal_ EnableSubnormals | _Literal_ RoundingMode | _Literal_ RoundingAccuracy
```

Comment:
I'm curious, why `A Type` operand is required? If I understand correctly, there should be no problems with inferring type of operand `A` (same for `B`)

Reply:
I spoke to Mikhail on Friday and I agree that the types for _A_ and _B_ should be inferrable from the context, so I'm going to remove them. New commit should be going up shortly.

Reply:
@mlychkov @AlexeySachkov should we make this change for the fixed point types as well (removing the input types)?

Reply:
Yes, I think that it is better to align this format of instructions for both specs.

Reply:
Done :)